### PR TITLE
implement live message for dashboard

### PR DESCRIPTION
**Pull request by Skylsmoi**
https://github.com/tracim/tracim/issues/2850

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

**Comment by Skylsmoi** [comment]
What to understand in this PR:
- First, check the changes in file Dashboard.jsx
  - it register the hanlers on the related eventType
- these handlers call an action creator to update redux
  - the action creator are the same used to add content in workspace content page (they already existed)
    - the action creators take an array of content in parameter, it explains the array in `[data.content]` from `this.props.dispatch(addWorkspaceContentList([data.content]))` because a TLM is for one content only
- Then, check the file action-creator.sync.js
  - the 2 new functions `addWorkspaceMember` and `updateWorkspaceMember` modify their parameters
    - This is to keep the `props.dispatch()` of the component simple
    - But we need the action creator to have that structure to allow the reducers to work as if it was a response of an endpoint
      - Because there was already a reducer implementing this action in reducer/workspaceList.js

File: frontend/src/reducer/currentWorkspace.js [hunk 1]
```diff
@@ -24,119 +26,129 @@ const defaultWorkspace = {
...
         memberList: [
           ...state.memberList,
-          {
-            id: action.newMember.user_id,
-            publicName: action.newMember.public_name,
-            role: action.role,
-            isActive: action.newMember.is_active,
-            doNotify: action.newMember.do_notify
-          }
+          { ...serializeMember(action.newMember) }
```
Comment (raphj) [inline frontend/src/reducer/currentWorkspace.js:85]:
```suggestion
          serializeMember(action.newMember)
```

File: frontend/test/src/container/Dashboard.spec.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          expect(setWorkspaceDetailSpy.called).to.equal(true)
+        })
+
+        it('should call this.setHeadTitle()', () => {
+          expect(setHeadTitleSpy.called).to.equal(true)
+        })
+      })
+    })
+
+    describe('eventType user_workspace_role', () => {
```
Comment (Skylsmoi) [inline frontend/test/src/container/Dashboard.spec.js:111]: rename to workspace_role

File: frontend/src/reducer/currentWorkspace.js [hunk 3]
```diff
@@ -10,8 +11,9 @@ import {
   WORKSPACE_MEMBER, UPDATE,
   USER_WORKSPACE_DO_NOTIFY,
   FOLDER_READ,
-  WORKSPACE_AGENDA_URL
+  WORKSPACE_AGENDA_URL, WORKSPACE_CONTENT
```
Comment (Skylsmoi) [inline frontend/src/reducer/currentWorkspace.js:14]: linting

File: frontend_lib/src/tracimLiveMessage.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ export const TLM_ENTITY_TYPE = {
   USER,
   CONTENT,
   SHAREDSPACE,
-  SHAREDSPACE_USER_ROLE
+  SHAREDSPACE_MEMBER
 }
 export const TLM_CORE_EVENT_TYPE = {
   CREATED,
```
Comment (Skylsmoi) [inline frontend_lib/src/tracimLiveMessage.js:19]: add undeleted

File: pushpin_cypress_config/pushpin.conf [hunk 5]
```diff
@@ -0,0 +1,145 @@
+[global]
+include={libdir}/internal.conf
+
+# directory to save runtime files
+rundir=/var/run/pushpin
+
+# prefix for zmq ipc specs
```
Comment (raphj) [inline pushpin_cypress_config/pushpin.conf:7]:
```suggestion
# prefix for ZeroMQ ipc specifications
```

File: pushpin_cypress_config/pushpin.conf [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+ipc_prefix=pushpin-
+
+# port offset for zmq tcp specs and http control server
```
Comment (raphj) [inline pushpin_cypress_config/pushpin.conf:10]:
```suggestion
# port offset for ZeroMQ tcp specifications and HTTP control server
```

File: pushpin_cypress_config/pushpin.conf [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+
+[runner]
+# services to start
+services=mongrel2,m2adapter,zurl,pushpin-proxy,pushpin-handler
+
+# plain HTTP port that mongrel2 should listen on
+http_port=7999
+
+# list of HTTPS ports that mongrel2 should listen on (you must have certs set)
```
Comment (raphj) [inline pushpin_cypress_config/pushpin.conf:24]:
```suggestion
# list of HTTPS ports that mongrel2 should listen on (certificates must be configured)
```

**Comment by Skylsmoi** [comment]
I need to wait for this PR to be merged https://github.com/tracim/tracim/pull/3101

then implement the change in TLM, fix tests then I'll be able to merge

**Comment by metzgegu** [comment]
Verified on branch.

**Pull request merged**